subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/ListenerBroadcastCodec.kt

```diff
@@ -24,16 +24,22 @@ import org.gradle.configurationcache.serialization.readCollection
 import org.gradle.configurationcache.serialization.writeCollection
 import org.gradle.internal.event.AnonymousListenerBroadcast
 import org.gradle.internal.event.ListenerManager
+import org.gradle.listener.ClosureBackedMethodInvocationDispatch
 
 
 internal
 class ListenerBroadcastCodec(private val listenerManager: ListenerManager) : Codec<AnonymousListenerBroadcast<*>> {
     override suspend fun WriteContext.encode(value: AnonymousListenerBroadcast<*>) {
         val broadcast: AnonymousListenerBroadcast<Any> = value.uncheckedCast()
-        writeClass(value.type)
+        val listenerType = value.type
+        writeClass(listenerType)
         val listeners = mutableListOf<Any>()
-        broadcast.visitListeners {
-            listeners.add(this)
+        broadcast.visitListenersUntyped {
+            val listener = this
+            if (isSupportedListener(listener, listenerType)) {
+                // TODO:configuration-cache consider emitting problems for unsupported listeners
```

Comment:
Question: Does this mean listeners that are not a supported type will be silently dropped when the broadcaster is serialized?  What kind of listeners would we lose here (i.e. which kind of listeners are unsupported)?

Reply:
Yes, this change set is reducing the set of listeners discarded  (now Groovy closures are handled), but still not covering all cases.

The listeners I saw being skipped in the JVM tests were all instances of `DefaultListenerManager.ListenerDispatch` - however, the type of the underlying type is the expected listener type. Will keep looking into that after this is merged.